“panic: unaligned 64-bit atomic operation” on 32-bit x86 systems

**Describe the bug**
If you compile Gitleaks using a 32-bit x86 version of Go, then `gitleaks git` will crash with this error: “panic: unaligned 64-bit atomic operation”.

**To Reproduce**
Steps to reproduce the behavior:

1. Make sure that you’re using a 32-bit x86 version of Go by running this command:

    ```
    go version
    ```

    If you’re using a correct version of Go, then that command’s output will end with `/386`. Example:

    <pre><samp>go version go1.23.4 linux/386</samp></pre>

1. Make sure that you have a local clone of the Gitleaks repo.

1. Change directory into your clone of the Gitleaks repo:

    ```
    cd <path-to-gitleaks-repo>
    ```

1. Build Gitleaks by running this command:

    ```
    make build
    ```

1. Try to use Gitleaks by running this command:

    ```
    ./gitleaks git
    ```

**Expected behavior**
I expect Gitleaks to work the same, regardless of whether its compiled for 32-bit x86 or x86-64.

**Basic Info (please complete the following information):**
 - OS: NixOS 24.11
 - Gitleaks Version: b022d1c856e13e2e4a1f559ea156b52986470e55

**Additional context**
Here’s a log for the crash:

<pre><samp>$ go version
go version go1.23.4 linux/386
$ cd repo/
$ make build
go fmt ./...
go mod tidy
go build -ldflags "-X=github.com/zricethezav/gitleaks/v8/cmd.Version=`git fetch --tags && git tag | sort -V | tail -1`"
$ ./gitleaks git

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

panic: unaligned 64-bit atomic operation

goroutine 67 [running]:
internal/runtime/atomic.panicUnaligned()
        /nix/store/b6iihna85jdb51pbq6j3q3v37l2q29m5-go-1.23.4/share/go/src/internal/runtime/atomic/unaligned.go:8 +0x2d
internal/runtime/atomic.Xadd64(0x958179c, 0x3da)
        /nix/store/b6iihna85jdb51pbq6j3q3v37l2q29m5-go-1.23.4/share/go/src/internal/runtime/atomic/atomic_386.s:125 +0x11
github.com/zricethezav/gitleaks/v8/detect.(*Detector).Detect(0x9581688, {{0x9b5f600, 0x3da}, {0x0, 0x0, 0x0}, {0xa6bd3e0, 0x10}, {0x0, 0x0}, ...})
        /home/jayman/Documents/Home/VC/Git/Not mine/gitleaks/repo/detect/detect.go:187 +0x57
github.com/zricethezav/gitleaks/v8/detect.(*Detector).DetectGit.func1()
        /home/jayman/Documents/Home/VC/Git/Not mine/gitleaks/repo/detect/git.go:51 +0x17b
github.com/fatih/semgroup.(*Group).Go.func1()
        /home/jayman/go/pkg/mod/github.com/fatih/semgroup@v1.2.0/semgroup.go:64 +0x76
created by github.com/fatih/semgroup.(*Group).Go in goroutine 1
        /home/jayman/go/pkg/mod/github.com/fatih/semgroup@v1.2.0/semgroup.go:60 +0x1fb
$ </samp></pre>

cc @zricethezav

<!--

Check out the Secret Scanning Discord Server for live discussions:

https://discord.gg/sydS6AHTUP

--->

--- a/cmd/root.go
+++ b/cmd/root.go
@@ -6,7 +6,6 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
-	"sync/atomic"
 	"time"
 
 	"github.com/rs/zerolog"
@@ -362,7 +361,7 @@ func bytesConvert(bytes uint64) string {
 }
 
 func findingSummaryAndExit(detector *detect.Detector, findings []report.Finding, exitCode int, start time.Time, err error) {
-	totalBytes := atomic.LoadUint64(&detector.TotalBytes)
+	totalBytes := detector.TotalBytes.Load()
 	bytesMsg := fmt.Sprintf("scanned ~%d bytes (%s)", totalBytes, bytesConvert(totalBytes))
 	if err == nil {
 		logging.Info().Msgf("%s in %s", bytesMsg, FormatDuration(time.Since(start)))

--- a/detect/detect.go
+++ b/detect/detect.go
@@ -89,7 +89,7 @@ type Detector struct {
 	ReportPath string
 	Reporter   report.Reporter
 
-	TotalBytes uint64
+	TotalBytes atomic.Uint64
 }
 
 // Fragment contains the data to be scanned
@@ -184,9 +184,9 @@ func (d *Detector) DetectString(content string) []report.Finding {
 // Detect scans the given fragment and returns a list of findings
 func (d *Detector) Detect(fragment Fragment) []report.Finding {
 	if fragment.Bytes == nil {
-		atomic.AddUint64(&d.TotalBytes, uint64(len(fragment.Raw)))
+		d.TotalBytes.Add(uint64(len(fragment.Raw)))
 	}
-	atomic.AddUint64(&d.TotalBytes, uint64(len(fragment.Bytes)))
+	d.TotalBytes.Add(uint64(len(fragment.Bytes)))
 
 	var findings []report.Finding
 

